src/app/history/git.js

```diff
@@ -59,16 +49,30 @@ export default class Git {
     }
   }
 
-  async filesInCommit(sha) {
-    const result = await this.git.raw('show', '--name-only', '--pretty=format:', sha);
-    return result.trim().split('\n');
-  }
-
   async isTracked(filepath) {
     const result = await this.git.raw('ls-files', this.relativePath(filepath));
     return !!result;
   }
 
+  async lastCommitAndFilePathMatchingGlob(glob) {
+    const [ latestCommit ] = await this.log([ '-n', '1', '--stat=4096', glob ]);
+
+    if (!latestCommit) {
+      return {};
+    }
+
+    const filePaths = latestCommit.diff.files.map(file => file.file);
+
+    if (filePaths.length > 1) {
+      throw new Error(`Only one document should have been recorded in ${latestCommit.hash}, but all these documens have been recorded: ${filePaths}`);
```

Comment:
I thought we had discovered this can happen without being a problem, for example when we manually rename files 🤔